go/pkg/client/cas_upload.go

```diff
@@ -607,3 +608,33 @@ func updateAndNotify(st *uploadState, bytesMoved int64, err error, missing bool)
 	st.clients = nil
 	st.ue = nil
 }
+
+// DetectArchiveUploads is an UploadCompressionClassifier that attempts to detect whether
+// uploads are already archives, either based on their initial few bytes or known file extensions.
+// It is a best-effort classifier that may not recognise every possible type.
+func DetectArchiveUploads(ue *uploadinfo.Entry) repb.Compressor_Value {
```

Comment:
I think this is very useful, but probably doesn't make a lot of sense to include it in the SDK. Different users might have different criteria and/or different performance requirements for this detection. Accommodating such variety becomes a bottleneck at the SDK level and is best left to the clients.

Reply:
I thought you might think that. I'll leave it out, although I feel like clients will need to reinvent this kind of logic in a few places. Still, maybe better not to add this prematurely.

Reply:
Were I not a maintainer I would definitely want this in the SDK, especially if someone smart already did the thinking for me :)
As a maintainer, I should anticipate this being a hot spot for PRs due to the large scope of implementation details.